### ISSUE TITLE
[error] Error: This socket has been ended by the other party

### ISSUE THREAD

**Issue by MeijeSibbel** [GooBox/goobox-community-gui#39]
```
[02:59:48.906] [verbose] Java HotSpot(TM) 64-Bit Server VM warning: Option CreateMinidumpOnCrash was deprecated in version 9.0 and will likely be removed in a future release. Use option CreateCoredumpOnCrash instead.
[02:59:49.302] [verbose] SLF4J: A number (1) of logging calls during the initialization phase have been intercepted and are
[02:59:49.302] [verbose] SLF4J: now being replayed. These are subject to the filtering rules of the underlying logging system.
[02:59:49.306] [verbose] SLF4J: See also http://www.slf4j.org/codes.html#replay
[02:59:49.506] [verbose] WARNING: An illegal reflective access operation has occurred
[02:59:49.506] [verbose] WARNING: Illegal reflective access by dorkbox.systemTray.util.SystemTrayFixes (file:/C:/Users/meije/Downloads/goobox-win32-x64%20(1)/goobox-win32-x64/resources/app/goobox-sync-storj/goobox-sync-storj-0.0.13.jar) to method java.lang.ClassLoader.findLoadedClass(java.lang.String)
[02:59:49.511] [verbose] WARNING: Please consider reporting this to the maintainers of dorkbox.systemTray.util.SystemTrayFixes
[02:59:49.530] [verbose] WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
[02:59:49.536] [verbose] WARNING: All illegal access operations will be denied in a future release
[03:00:10.638] [error] Cannot parse #: SyntaxError: Unexpected token # in JSON at position 0
[03:00:25.354] [error] Cannot parse #: SyntaxError: Unexpected token # in JSON at position 0
[03:00:28.460] [info] logging in to Storj: meije@storj.io
[03:00:28.460] [debug] sending a request to sync storj: {"method":"login","args":{"email":"meije@storj.io","password":"","encryptionKey":""}}

[03:00:28.463] [error] Error: This socket has been ended by the other party
    at Socket.writeAfterFIN [as write] (net.js:305:12)
    at Promise.race.resolve (C:\Users\meije\Downloads\goobox-win32-x64 (1)\goobox-win32-x64\resources\app\lib\main-process\startup.js:21099:18)
    at Promise (<anonymous>)
    at Storj.login (C:\Users\meije\Downloads\goobox-win32-x64 (1)\goobox-win32-x64\resources\app\lib\main-process\startup.js:21086:26)
    at EventEmitter._electron.ipcMain.on (C:\Users\meije\Downloads\goobox-win32-x64 (1)\goobox-win32-x64\resources\app\lib\main-process\startup.js:20575:26)
    at emitTwo (events.js:106:13)
    at EventEmitter.emit (events.js:194:7)
    at WebContents.<anonymous> (C:\Users\meije\Downloads\goobox-win32-x64 (1)\goobox-win32-x64\resources\electron.asar\browser\api\web-contents.js:247:13)
    at emitTwo (events.js:106:13)
    at WebContents.emit (events.js:194:7)

[03:00:29.680] [error] { code: 'EPIPE' }
```

How to reproduce: click on finish while the app is thinking.

![image](https://user-images.githubusercontent.com/11735738/34752445-b29b1e70-f5b2-11e7-962d-eff7869eb719.png)

**Comment by MeijeSibbel** [comment]
To reproduce: After successful login click back and try to login again.

**Comment by jkawamoto** [comment]
Since the above log and screen shot show passwords, I modified the original post.

**Comment by kaloyan-raev** [comment]
I believe this is caused by a multi-threading issue in the JNI code of java-libstorj that crashes the sync storj app. It has been reported also for the Android binding: https://github.com/Storj/android-libstorj/issues/15

I am currently working on a solution.

The workaround is to delete the auth file from the app data dir before importing the keys.

**Comment by kaloyan-raev** [comment]
Fixed in java-libstorj 0.5.1: https://github.com/Storj/java-libstorj/commit/f59e514ed1ac0a6a8d8e4ae19ecd10446e4ba608
